code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -704,17 +706,16 @@ var/global/list/obj/item/device/pda/PDAs = list()
 		return
 
 	if (usr.canUseTopic(src))
-		var/obj/item/weapon/pen/O = locate() in src
-		if(O)
-			if (istype(loc, /mob))
-				var/mob/M = loc
-				if(M.get_active_hand() == null)
-					M.put_in_hands(O)
-					usr << "<span class='notice'>You remove \the [O] from \the [src].</span>"
+		if(is_type_in_list(inserted_item, contained_item))

```

Comment:
no point in checking this, if the item is inserted in the pda it must be one of those types.